archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -960,6 +962,24 @@ public JavaConstructor getConstructor(String... parameters) {
         return javaClassDependencies.getDirectDependenciesFromClass();
     }
 
+    /**
+     * Returns the transitive closure of all dependencies originating from this class, i.e. its direct dependencies
+     * and the dependencies from all target classes that are within the given package (including sub-packages).
+     * Note that all target classes in this package must be imported.
+     * @param packageNamePrefix start of the package name of classes whose transitive dependencies should be considered
+     * @return all transitive dependencies (including direct dependencies) from this class
+     * @see #getDirectDependenciesFromSelf()
+     */
+    @PublicAPI(usage = ACCESS)
+    public Set<Dependency> getTransitiveDependenciesFromSelfWithinPackage(final String packageNamePrefix) {
```

Comment:
To me `WithinPackage` still seems like a rather arbitrary restriction :wink: What about two public methods

```
Set<Dependency> getTransitiveDependenciesFromSelf();
Set<Dependency> getTransitiveDependenciesFromSelf(DescribedPredicate<JavaClass> incudePredicate);
```
?

Reply:
To keep it simple & stupid, I'm actually inclined to drop the version with the `includePredicate` again.
I had used the package-restriction initially only to work around #397, but as this could be solved quite easily, the filtering is IMO not needed anymore.

Reply:
I've updated the PR accordingly.